### PR TITLE
Add readme and sample project

### PR DESCRIPTION
**Pull request by carlosedp**
This PR adds a readme with configuration samples and an example project using the plugin.

### CODE REVIEW

**Review by lefou — COMMENTED** [review]
Looks good to me. The setup uses Apache-2 as a license for publishing. Would be nice to have it as `LICENSE` file in the repo.

File: Readme.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import $ivy.`io.github.alexarchambault.mill::mill-native-image::0.1.23`
+import io.github.alexarchambault.millnativeimage.NativeImage
+```
+
+## Usage
+
+Sample configuration:
+
+```scala
+import mill._, mill.scalalib._, mill.scalalib.scalafmt._
```
[the remark this answers is not in the document]
Comment (carlosedp) [reply Readme.md:21]: Good catch!

File: example/build.sc [hunk 2]
```diff
@@ -0,0 +1,70 @@
+import mill._, mill.scalalib._, mill.scalalib.scalafmt._
```
[the remark this answers is not in the document]
Comment (carlosedp) [reply example/build.sc:1]: 👍🏻

File: Readme.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
+# Mill-native-image Scala Mill Plugin
+
+This is a [Mill](https://com-lihaoyi.github.io/mill/mill/Intro_to_Mill.html) plugin which allows building executables using GraalVM native-images for your modules.
```
Comment (alexarchambault) [inline Readme.md:3]:
```suggestion
This is a [Mill](https://com-lihaoyi.github.io/mill/mill/Intro_to_Mill.html) plugin which allows building executables using [GraalVM native-image](https://www.graalvm.org/latest/reference-manual/native-image/) for your modules.
```
(minor typo, and adds a link to the GraalVM website)

**Review by alexarchambault — APPROVED** [review]
Thanks @carlosedp! There would have been a few suggestions or minor things I would have liked to fix in the README, but I don't have enough time for it immediately, so I'm fine merging as is, this is still much better than what we had before.

**Comment by alexarchambault** [comment]
And thanks @lefou for the review

**Pull request merged**